From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Tue, 13 Aug 2019 07:46:10 -0400
Subject: [PATCH 2/2] Manipulate PATH directly instead of _call_ing conda.bat
Updated: 2022-12-14 by @jaimergp, conda 22.11.1 compatibility
Updated: 2023-06-21 by @jaimergp, conda 23.5.0 compatibility

diff --git a/conda/utils.py b/conda/utils.py
index bc340fb10..d6b8a208d 100644
--- a/conda/utils.py
+++ b/conda/utils.py
@@ -419,7 +419,16 @@ def wrap_subprocess_call(
             # after all!
             # fh.write("@FOR /F \"tokens=100\" %%F IN ('chcp') DO @SET CONDA_OLD_CHCP=%%F\n")
             # fh.write('@chcp 65001>NUL\n')
-            fh.write(f'{silencer}CALL "{conda_bat}" activate "{prefix}"\n')
+            # fh.write(f'{silencer}CALL "{conda_bat}" activate "{prefix}"\n')
+            fh.write(
+                f'SET "PATH={prefix};'
+                f'{prefix}\\Library\\mingw-w64\\bin;'
+                f'{prefix}\\Library\\usr\\bin;'
+                f'{prefix}\\Library\\bin;'
+                f'{prefix}\\Scripts;'
+                f'{prefix}\\bin;'
+                '%PATH%"'
+            )
             fh.write(f"{silencer}IF %ERRORLEVEL% NEQ 0 EXIT /b %ERRORLEVEL%\n")
             if debug_wrapper_scripts:
                 fh.write("echo *** environment after *** 1>&2\n")
